src/endpoint-client.ts

```diff
@@ -60,6 +60,22 @@ export interface EndpointClientRequestOptions <T> {
 	dryRunReturnValue?: T
 }
 
+/**
+ * Convert to string and scrub sensitive values like auth tokens
+ * Meant to be used before logging the request
+ */
+function scrubConfig(config: AxiosRequestConfig): string {
+	const message = JSON.stringify(config)
+	const bearerRegex = /"(Bearer )([0-9a-f]{8})[0-9a-f-]{28}"/i
```

Comment:
Obviously not a big deal (and maybe you like it better the way you have it) but the first two capture groups could just be one (`/"(Bearer [0-9a-f]{8})[0-9a-f-]{28}"/i`) and then you could just use `$1` in the replace below.